resources/validations/ui/src/context/browser/presenter/router.ts

```diff
@@ -0,0 +1,112 @@
+import { match } from 'path-to-regexp';
+
+export type RouteTypes = {
+  Home: { type: 'Home' };
+  Summary: { type: 'Summary' };
+  Assertion: {
+    type: 'Assertion';
+    assertionId: string;
+  };
+};
+
+export type Route = RouteTypes[keyof RouteTypes];
+export type RouteType = Route['type'];
+export namespace Routes {
+  export const home: RouteTypes['Home'] = {
+    type: 'Home',
+  };
+  export const summary: RouteTypes['Summary'] = {
+    type: 'Summary',
+  };
+  export const assertion = (options: {
+    assertionId: string;
+  }): RouteTypes['Assertion'] => {
+    return {
+      type: 'Assertion',
+      assertionId: options.assertionId,
+    };
+  };
+}
+
+export type NotFound = { type: 'NotFound' };
+export const notFound: NotFound = { type: 'NotFound' };
+
+const RouteUrl: Record<Route['type'], (route?: any) => string> = {
+  Home: () => '#/',
+  Summary: () => '#/summary',
+  Assertion: (route: RouteTypes['Assertion']) =>
+    `#/assertions/${route.assertionId}`,
+};
+
+export const getUrl = (route: Route): string => {
+  return RouteUrl[route.type](route);
+};
+
+const matchRoute = <L extends Route>(
+  urlPattern: string,
+  createRoute: (params?: any) => L,
+) => {
+  const matcher = match(urlPattern);
+  return (url: string) => {
+    const match = matcher(url);
+    if (match) {
+      return createRoute(match.params);
+    }
+  };
+};
+
+const RouteMatch: Record<Route['type'], (url: string) => Route | undefined> = {
+  Home: matchRoute('#/', () => Routes.home),
+  Summary: matchRoute('#/summary', () => Routes.summary),
+  Assertion: matchRoute('#/assertions/:assertionId', Routes.assertion),
+};
+
+export const getRoute = (url: string): Route | NotFound => {
+  for (const routeType of Object.keys(RouteMatch)) {
+    const route = RouteMatch[routeType as Route['type']](url);
+    if (route) {
+      return route;
+    }
+  }
+  return notFound;
+};
+
+type Breadcrumb = {
+  text: string;
+  url: string;
+  selected: boolean;
+};
+export const breadcrumbs: Record<Route['type'], (route: any) => Breadcrumb[]> =
+  {
+    Home: (route: Route) => {
+      return [
+        {
+          text: 'Select SSP',
+          url: getUrl(Routes.home),
+          selected: route.type === 'Home',
+        },
+      ];
+    },
+    Summary: (route: Route) => {
+      return [
+        ...breadcrumbs.Home(route),
+        {
+          text: 'Document name',
+          url: getUrl(Routes.summary),
+          selected: route.type === 'Summary',
+        },
+      ];
+    },
+    Assertion: (route: RouteTypes['Assertion']) => {
+      return [
+        ...breadcrumbs.Home(route),
+        {
+          text: 'Assertion',
+          url: getUrl(Routes.assertion({ assertionId: route.assertionId })),
+          selected: route.type === 'Assertion',
+        },
+      ];
+    },
+  };
```

Comment:
B: I think we should talk with @rrkane10x tomorrow, but I am not sure what we ought to do with the breadcrumbs, but, for example, when looking at the SSP context, the ability to go back to the error listing is not 100% obvious to me given the text.

<img width="1167" alt="Screen Shot 2021-07-07 at 4 19 38 PM" src="https://user-images.githubusercontent.com/61464190/124830091-6c71c700-df47-11eb-9164-12ea440be5f6.png">

Reply:
I don't know where that separator character in your screenshot is coming from - should look like this 
![image](https://user-images.githubusercontent.com/136512/124833179-6548ba00-df43-11eb-90f0-5e2abe69135f.png)
what browser are you on?

Reply:
The artifact not so much a big deal, I think I had the mouse hovering over it. I mean just the wording itself `Select SSP > Assertion` either doesn't convey to me that it is a breadcrumb and that you can move up a level. Not sure wording or visual effects can help that.